### PR TITLE
Return LSP 3.17 labelDetails on completion items

### PR DESCRIPTION
**Pull request by lukaszsamson**
annotate completions that require alias insertions deprioretize alias insertion
fix crash when completing one line source file

Examples from vscode
Screenshot 2022-12-13 at 07 01 19
Screenshot 2022-12-13 at 07 01 43
Screenshot 2022-12-13 at 07 02 16

Addresses https://github.com/elixir-lsp/elixir-ls/issues/774

### CODE REVIEW

**Comment by scohen** [comment]
I like the `alias XXX` annotation, that fixes the earlier problems.

Slight nitpick though: It's puzzling to me why typing `Enum.` would have `EmptyError` and `OutOfBoundsException` show up before `Enum.all`.  Exceptions should be sorted pretty far down in the list of possible completions, but this does bring up structs, which are probably as important as functions, but shouldn't be super high on the list for this specific completion.

File: apps/language_server/lib/language_server/providers/completion.ex [hunk 1]
```diff
@@ -338,12 +345,23 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
         range: range(line_to_insert_alias, 0, line_to_insert_alias, 0),
         newText: alias_edit
       },
-      documentation: alias_value <> "\n" <> summary
+      documentation: alias_value <> "\n" <> summary,
+      label_details:
+        completion_without_additional_text_edit.label_details
+        |> Map.update!("description", &("alias " <> &1)),
```
Comment (scohen) [inline apps/language_server/lib/language_server/providers/completion.ex:351]: nit: don't use a pipe here.

File: apps/language_server/lib/language_server/source_file.ex [hunk 2]
```diff
@@ -334,9 +334,12 @@ defmodule ElixirLS.LanguageServer.SourceFile do
...
 
   def elixir_position_to_lsp(urf8_text, {elixir_line, elixir_character}) do
-    utf16_character =
+    line =
       lines(urf8_text)
       |> Enum.at(max(elixir_line - 1, 0))
+
+    utf16_character =
+      (line || "")
       |> elixir_character_to_lsp(elixir_character)
```
Comment (scohen) [inline apps/language_server/lib/language_server/source_file.ex:343]: i really think this is being made less clear due to the pipes
```suggestion
    utf16_character = elixir_character_to_lsp(line || "", elixir_character)
```

**Comment by lukaszsamson** [comment]
> Slight nitpick though: It's puzzling to me why typing Enum. would have EmptyError and OutOfBoundsException show up before Enum.all. Exceptions should be sorted pretty far down in the list of possible completions, but this does bring up structs, which are probably as important as functions, but shouldn't be super high on the list for this specific completion.

Nice idea. It was easy to deprioretize exceptions.

Screenshot 2022-12-14 at 23 40 48

**Review by scohen — APPROVED** [review]

**Pull request merged**